frontend/src/metabase/dashboard/components/add-card-sidebar/QuestionList.jsx

```diff
@@ -0,0 +1,70 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { t } from "ttag";
+
+import EmptyState from "metabase/components/EmptyState";
+import Search from "metabase/entities/search";
+
+import { QuestionPickerItem } from "./QuestionPickerItem";
+
+import { EmptyStateContainer } from "./QuestionList.styled";
+
+QuestionList.propTypes = {
+  searchText: PropTypes.string,
+  collectionId: PropTypes.string,
+  onSelect: PropTypes.func.isRequired,
+  hasCollections: PropTypes.bool,
+};
+
+export function QuestionList({
+  searchText,
+  collectionId,
+  onSelect,
+  hasCollections,
+}) {
+  if (collectionId === "personal" && !searchText) {
+    return null;
+  }
+
+  const trimmedSearchText = searchText.trim();
+  const isSearching = !!trimmedSearchText;
+
+  let query = isSearching
+    ? { q: trimmedSearchText }
+    : { collection: collectionId };
+
+  query = {
+    ...query,
+    models: "card",
+  };
+
+  return (
+    <Search.ListLoader entityQuery={query} wrapped>
+      {({ list }) => {
+        const shouldShowEmptyState =
+          list.length === 0 && (isSearching || !hasCollections);
+        if (shouldShowEmptyState) {
+          return (
+            <EmptyStateContainer>
+              <EmptyState message={t`Nothing here`} icon="all" />
+            </EmptyStateContainer>
+          );
+        }
+
+        return (
+          <ul>
```

Comment:
If you take the second approach listed below (adding `role="menuitem"` to the `li` tags) you will also need to add `role="menu"` to this `ul`